samples/core/iris/iris_pipeline.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Iris flowers example using TFX. Fork from https://github.com/tensorflow/tfx/blob/master/tfx/examples/iris/iris_pipeline_native_keras.py"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import kfp
+from typing import Text
+
+import absl
+import tensorflow_model_analysis as tfma
+
+from tfx.components import CsvExampleGen
+from tfx.components import Evaluator
+from tfx.components import ExampleValidator
+from tfx.components import SchemaGen
+from tfx.components import StatisticsGen
+from tfx.components import Trainer
+from tfx.components import Transform
+from tfx.components.base import executor_spec
+from tfx.components.trainer.executor import GenericExecutor
+from tfx.orchestration import data_types
+from tfx.orchestration import pipeline
+from tfx.orchestration.kubeflow import kubeflow_dag_runner
+from tfx.proto import trainer_pb2
+from tfx.utils.dsl_utils import external_input
+
+_pipeline_name = 'iris_native_keras'
+
+# This example assumes that Iris flowers data is stored in GCS and the
+# utility function is in iris_utils.py. Feel free to customize as needed.
+_data_root_param = data_types.RuntimeParameter(
+    name='data-root',
+    default='gs://ml-pipeline-playground/iris/data',
+    ptype=Text,
+)
+
+# Python module file to inject customized logic into the TFX components. The
+# Transform and Trainer both require user-defined functions to run successfully.
+_module_file_param = data_types.RuntimeParameter(
+    name='module-file',
+    default='gs://ml-pipeline-playground/iris/modules/iris_utils_native_keras.py',
+    ptype=Text,
+)
+
+# Directory and data locations. This example assumes all of the flowers
+# example code and metadata library is relative to a GCS path.
+_pipeline_root = os.path.join(
+    'gs://jxzheng-helloworld', 'tfx_iris', kfp.dsl.RUN_ID_PLACEHOLDER
```

Comment:
Can we use the "gs://<your-gcs-bucket>/" placeholder here?

Reply:
Will update this PR after today's bugbash. I want to make sure that magic works :)

Reply:
Okay it only partially works :- ( 

Fixed the backend in #3145